libV2/helpers/collection/generateSkeletionTreeFromOpenAPI.js

```diff
@@ -0,0 +1,261 @@
+let _ = require('lodash'),
+  Graph = require('graphlib').Graph,
+
+  PATH_WEBHOOK = 'path~webhook',
+  ALLOWED_HTTP_METHODS = {
+    get: true,
+    head: true,
+    post: true,
+    put: true,
+    patch: true,
+    delete: true,
+    connect: true,
+    options: true,
+    trace: true
+  },
+
+  _generateTreeFromPaths = function (openapi, { includeDeprecated }) {
+    /**
+     * We will create a unidirectional graph
+     */
+    let tree = new Graph();
+
+    tree.setNode('root:collection', {
+      type: 'collection',
+      data: {},
+      meta: {}
+    });
+
+    _.forEach(openapi.paths, function (methods, path) {
+      let pathSplit = path === '/' ? [path] : _.compact(path.split('/'));
+
+      // if after path split we just have one entry
+      // that means no folders need to be generated.
+      // check for all the methods inside it and expand.
+      if (pathSplit.length === 1) {
```

Comment:
We do need to generate folder when there's one path segment. As for 2 path segments we generate 2 folders but for 1, no folders are generated ATM.